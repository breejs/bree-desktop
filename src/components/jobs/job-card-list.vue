<script setup>
import JobStatus from './job-status.vue';
import JobInfo from './job-info.vue';

const props = defineProps({
  jobs: {
    type: Array,
    required: true
  }
});
</script>

<template lang="pug">
template(v-if='jobs.length > 0')
  .card.w-100.mb-3(v-for='job in props.jobs', :key='job.name')
    .card-body.d-flex.flex-row
      .status.me-3.d-flex.align-items-center
        JobStatus(:status='job.status')
      .info.d-flex.flex-column.align-self-center
        JobInfo(:name='job.name', :lastRun='job.lastRun')
template(v-else)
  p= 'No jobs to show'
</template>
