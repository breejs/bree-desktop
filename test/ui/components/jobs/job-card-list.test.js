import { storyTests } from '../../_story-tests';

import * as stories from '@/components/jobs/job-card-list.stories';

storyTests(stories);
