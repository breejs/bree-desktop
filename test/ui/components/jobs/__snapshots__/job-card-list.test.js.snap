// Vitest Snapshot v1

exports[`story snapshots > Jobs 1`] = `
"<div class=\\"card w-100 mb-3\\">
  <div class=\\"card-body d-flex flex-row\\">
    <div class=\\"status me-3 d-flex align-items-center\\"><i class=\\"bi bi-circle-fill text-info\\"></i></div>
    <div class=\\"info d-flex flex-column align-self-center\\">
      <div class=\\"row\\">
        <h4>Waiting Job</h4>
      </div>
      <div class=\\"row\\">
        <p class=\\"text-muted\\">Last run: 4/4/2022, 12:00:00 PM</p>
      </div>
    </div>
  </div>
</div>
<div class=\\"card w-100 mb-3\\">
  <div class=\\"card-body d-flex flex-row\\">
    <div class=\\"status me-3 d-flex align-items-center\\"><i class=\\"bi bi-circle-fill text-warning\\"></i></div>
    <div class=\\"info d-flex flex-column align-self-center\\">
      <div class=\\"row\\">
        <h4>Stopped Job</h4>
      </div>
      <div class=\\"row\\">
        <p class=\\"text-muted\\">Last run: 4/4/2022, 5:00:00 AM</p>
      </div>
    </div>
  </div>
</div>
<div class=\\"card w-100 mb-3\\">
  <div class=\\"card-body d-flex flex-row\\">
    <div class=\\"status me-3 d-flex align-items-center\\"><i class=\\"bi bi-circle-fill text-danger\\"></i></div>
    <div class=\\"info d-flex flex-column align-self-center\\">
      <div class=\\"row\\">
        <h4>Errored Job</h4>
      </div>
      <div class=\\"row\\">
        <p class=\\"text-muted\\">Last run: never</p>
      </div>
    </div>
  </div>
</div>
<div class=\\"card w-100 mb-3\\">
  <div class=\\"card-body d-flex flex-row\\">
    <div class=\\"status me-3 d-flex align-items-center\\"><i class=\\"bi bi-circle-fill text-success\\"></i></div>
    <div class=\\"info d-flex flex-column align-self-center\\">
      <div class=\\"row\\">
        <h4>Running Job</h4>
      </div>
      <div class=\\"row\\">
        <p class=\\"text-muted\\">Last run: 4/4/2022, 8:00:00 AM</p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`story snapshots > NoJobs 1`] = `"<p>No jobs to show</p>"`;

exports[`story snapshots > WithLastRun 1`] = `
"<div class=\\"row\\">
  <h4>Job Name</h4>
</div>
<div class=\\"row\\">
  <p class=\\"text-muted\\">Last run: 4/4/2022, 12:00:00 PM</p>
</div>"
`;

exports[`story snapshots > WithNoLastRun 1`] = `
"<div class=\\"row\\">
  <h4>Job Name</h4>
</div>
<div class=\\"row\\">
  <p class=\\"text-muted\\">Last run: never</p>
</div>"
`;
